apps/src/studio/studio.js

```diff
@@ -5353,6 +5353,11 @@ Studio.queueCallback = function (callback, args) {
   };
 
   registerEventHandler(Studio.eventHandlers, handlerName, () => {
+    // remove the last argument because stepCallExpression always wants to push it back on.
+    if (state.arguments.length > 0) {
+      state.value = state.arguments.pop();
+    }
+
     const depth = Studio.interpreter.stateStack.unshift(state);
```

Comment:
Yeah this code is super gross 😕.  I was hoping we could replace the immediate callback with `await` or `Promise` queues but it's not on the horizon until after back-to-school.